dtfs-central-api/api-tests/v1/tfm/deal-cancellation/tfm-deal-cancellation-post.api-test.ts

```diff
@@ -0,0 +1,156 @@
+  }) as AnyObject;
+
+  beforeAll(async () => {
+    await wipeDB.wipe([MONGO_DB_COLLECTIONS.TFM_DEALS, MONGO_DB_COLLECTIONS.DEALS]);
+  });
+
+  beforeEach(async () => {
+    const createDealResponse: { body: { _id: string } } = await createDeal({ deal: newDeal, user: aPortalUser() });
+    dealId = createDealResponse.body._id;
+    tfmUserId = aTfmUser()._id;
+    auditDetails = generateTfmAuditDetails(tfmUserId);
+    submitDealCancellationUrl = `/v1/tfm/deals/${dealId}/cancellation/submit`;
+
+    await testApi
+      .put({
+        dealType: DEALS.DEAL_TYPE.BSS_EWCS,
+        dealId,
+        submissionType: DEALS.SUBMISSION_TYPE.AIN,
+        auditDetails: generatePortalAuditDetails(MOCK_PORTAL_USER._id),
+      })
+      .to('/v1/tfm/deals/submit');
+
+    await testApi
+      .put({
+        dealUpdate: {
+          tfm: {
+            dateReceived: '23-09-2024',
+            dateReceivedTimestamp: 1727085149,
+            parties: {},
+            activities: [],
+            product: DEALS.DEAL_TYPE.BSS_EWCS,
+            stage: TFM_DEAL_STAGE.CONFIRMED,
+            exporterCreditRating: 'Acceptable (B+)',
+            lastUpdated: 1727085149571,
+            cancellation,
+            lossGivenDefault: 50,
+            probabilityOfDefault: 12,
+          },
+        },
+        auditDetails: generatePortalAuditDetails(MOCK_PORTAL_USER._id),
+      })
+      .to(`/v1/tfm/deals/${dealId}`);
+  });
+
+  afterEach(async () => {
+    process.env = { ...originalProcessEnv };
+    await wipeDB.wipe([MONGO_DB_COLLECTIONS.TFM_DEALS, MONGO_DB_COLLECTIONS.DEALS]);
+  });
+
+  describe('POST /v1/tfm/deals/:dealId/cancellation/submit', () => {
+    describe('when FF_TFM_DEAL_CANCELLATION_ENABLED is disabled', () => {
+      beforeEach(() => {
+        process.env.FF_TFM_DEAL_CANCELLATION_ENABLED = 'false';
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+      });
+
+      it('should return 404', async () => {
+        const { status } = await testApi.post({ cancellation, auditDetails }).to(submitDealCancellationUrl);
+
+        expect(status).toEqual(HttpStatusCode.NotFound);
+      });
+    });
+
+    describe('when FF_TFM_DEAL_CANCELLATION_ENABLED is enabled', () => {
+      beforeEach(() => {
+        process.env.FF_TFM_DEAL_CANCELLATION_ENABLED = 'true';
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+      });
+
+      withMongoIdPathParameterValidationTests({
+        baseUrl: '/v1/tfm/deals/:dealId/cancellation/submit',
+        makeRequest: (url) => testApi.post({ cancellation, auditDetails }).to(url),
+      });
+
+      it('should return the submit cancellation response object if a matching deal and cancellation exists', async () => {
+        const submitCancellationResponse = await testApi.post({ cancellation, auditDetails }).to(submitDealCancellationUrl);
+
+        expect(submitCancellationResponse.body).toEqual({ cancelledDeal: { ukefDealId: dealId } });
```

Comment:
It maybe makes more sense to just return the entire deal object here? Not a big deal as this is all I need for the emails

Reply:
Have changed the structure now (following offline discussion) to 
```
export type TfmDealCancellationResponse = {
  cancelledDealUkefId: string | ObjectId;
  riskExpiredFacilityUkefIds?: (string | ObjectId)[];
};
```